r/configure

```diff
@@ -175,12 +165,6 @@ find_arrow () {
     # 2. Use pkg-config to find arrow on the system
     _LIBARROW_FOUND="`${PKG_CONFIG} --variable=prefix --silence-errors ${PKG_CONFIG_NAME}`"
     echo "*** Trying Arrow C++ found by pkg-config: $_LIBARROW_FOUND"
-  elif brew --prefix ${PKG_BREW_NAME} > /dev/null 2>&1; then
-    # 3. On macOS, look for Homebrew apache-arrow
-    #    (note that if you have pkg-config, homebrew arrow may have already been found)
-    _LIBARROW_FOUND=`brew --prefix ${PKG_BREW_NAME}`
-    echo "*** Trying Arrow C++ found by Homebrew: ${_LIBARROW_FOUND}"
-    export PKG_CONFIG_PATH="${_LIBARROW_FOUND}/lib/pkgconfig${PKG_CONFIG_PATH:+:${PKG_CONFIG_PATH}}"
```

Comment:
Ah, no (though I think it is a rather rare scenario due to CRAN hosting mac binaries...)

Reply:
The argument for keeping Homebrew support would be that `brew install apache-arrow --HEAD` can be used (often) to enable R package development. Unless we do something else (like download the latest C++ binary if the version number ends in .9000), this could be a step backwards for the casual contributor on macOS.

Reply:
I'm not sure I agree that this is a pattern we want to get rid of so quickly (I've onboarded a number of new contributors, and if they've got a purely R fix,`brew install apache-arrow --HEAD` was the quickest way forward.

But if this is removed, we should also update the docs at https://arrow.apache.org/docs/r/articles/developers/setup.html (and anywhere else we've listed that)

Reply:
I think we should add a mechanism to get the latest nightly binary when the dev version is .9000 and handle that os agnostic (iirc there is a issues open for that already). If that doesn't work due to incompatibility we/user can just build from source (e.g. via `LIBARROW_BINARY=FALSE`). On a git checkout this would need `make sync-cpp` but I want to integrate that into normal {pkgbuild} path via bootstrap.R soon.

But I can also add this back for now :shrug: